pkgs/by-name/xo/xonsh/package.nix

```diff
@@ -1,23 +1,114 @@
-{ lib
-, callPackage
-, extraPackages ? (ps: [ ])
-, runCommand
+{
+  lib,
+  callPackage,
+  coreutils,
+  fetchFromGitHub,
+  git,
+  gitUpdater,
+  glibcLocales,
+  python3Packages,
 }:
 
 let
-  xonsh-unwrapped = callPackage ./unwrapped.nix { };
-  inherit (xonsh-unwrapped.passthru) python;
 
-  pythonEnv = python.withPackages (ps: [
-    (ps.toPythonModule xonsh-unwrapped)
-  ] ++ extraPackages ps);
+  argset = {
+    pname = "xonsh";
+    version = "unstable-2024-05-25";
+    pyproject = true;
+
+    # PyPI package ships incomplete tests
+    src = fetchFromGitHub {
+      owner = "xonsh";
+      repo = "xonsh";
+      # rev = "refs/tags/${argset.version}";
+      rev = "1613914d507884487251ecafb55e281699931eb8";
+      hash = "sha256-CSElfkeQZi8o8H9WSZtCvzfk6qCSGEC9ctkI0sCBcSs=";
+    };
+
+    nativeBuildInputs = with python3Packages; [
+      setuptools
+      wheel
+    ];
+
+    propagatedBuildInputs = (with python3Packages; [
+      ply
+      prompt-toolkit
+      pygments
+    ]);
+
+    nativeCheckInputs = [
+      git
+      glibcLocales
+    ] ++ (with python3Packages; [
+      pip
+      pyte
+      pytest-mock
+      pytest-subprocess
+      pytestCheckHook
+      requests
+    ]);
+
+    disabledTests = [
+      # fails on sandbox
+      "test_colorize_file"
+      "test_loading_correctly"
+      "test_no_command_path_completion"
+      "test_bsd_man_page_completions"
+      "test_xonsh_activator"
+      "test_subproc_output_format"
+      "test_raise_subproc_error_with_show_traceback"
```

Comment:
These are 2 new tests that are failing. Not sure if they should be added here or under "fails on non-interactive shells", or be reported to upstream.

Reply:
`test_subproc_output_format` is marked as a flaky test. Disabling all the flaky tests now.